code/modules/projectiles/pins.dm

```diff
@@ -0,0 +1,189 @@
+/obj/item/device/firing_pin
+	name = "electronic firing pin"
+	desc = "A small authentication device, to be inserted into a firearm receiver to allow operation. NT safety regulations require all new designs to incorporate one."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "firing_pin"
+	item_state = "pen"
+	origin_tech = "materials=2;combat=4"
+	flags = CONDUCT
+	w_class = 1
+	attack_verb = list("poked")
+	var/emagged = FALSE
+	var/fail_message = "<span class='warning'>INVALID USER.</span>"
+	var/selfdestruct = 0 // Explode when user check is failed.
+	var/force_replace = 0 // Can forcefully replace other pins.
+	var/pin_removeable = 0 // Can be replaced by any pin.
+	var/obj/item/weapon/gun/gun
+
+
+/obj/item/device/firing_pin/New(newloc)
```

Comment:
Initialize.

Reply:
You also should add a Destroy() proc here that nulls out `gun`.

Reply:
I think that 
`/obj/item/device/firing_pin/Destroy()
	if(gun)
		gun.pin = null
	return ..()`

Handles that, but I'm not quite sure.